### PR TITLE
Fix panic when calling UpdateStatus if LastTransitionTime no set

### LINKED ISSUE #890: failed to create job (job-type: mpi)when use volcano v.4.0
Hi everyone,
I am trying to create a mpi type job , but failed, and there are some errors in vc controller log.

volcano version: v0.4.0

job yaml： almost same with https://github.com/volcano-sh/volcano/tree/master/example/kubecon-2019-china/mpi-sample

error log：
```
E0630 08:41:42.001113       1 job_controller_actions.go:548] Failed to update status of Job default/lm-mpi-job: Job.batch.volcano.sh "lm-mpi-job" is invalid: status.state.lastTransitionTime: Invalid value: "null": status.state.lastTransitionTime in body must be of type string: "null"
I0630 08:41:42.001195       1 panic.go:679] Finished Job <default/lm-mpi-job> initiate
I0630 08:41:42.001200       1 panic.go:679] Finished Job <default/lm-mpi-job> sync up
E0630 08:41:42.001276       1 runtime.go:78] Observed a panic: "invalid memory address or nil pointer dereference" (runtime error: invalid memory address or nil pointer dereference)
goroutine 431 [running]:
k8s.io/apimachinery/pkg/util/runtime.logPanic(0x13c30e0, 0x21ba5c0)
	/mnt/go/pkg/mod/k8s.io/apimachinery@v0.16.9-beta.0/pkg/util/runtime/runtime.go:74 +0xa3
k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
	/mnt/go/pkg/mod/k8s.io/apimachinery@v0.16.9-beta.0/pkg/util/runtime/runtime.go:48 +0x82
panic(0x13c30e0, 0x21ba5c0)
	/usr/local/go/src/runtime/panic.go:679 +0x1b2
volcano.sh/volcano/pkg/apis/batch/v1alpha1.(*Job).GetObjectKind(0x0, 0x17c7ac0, 0x0)
	<autogenerated>:1 +0x5
k8s.io/client-go/tools/reference.GetReference(0xc0002f5570, 0x176edc0, 0x0, 0x4dbf0b, 0x7f8eda6fc7e0, 0xc000c78918)
	/mnt/go/pkg/mod/k8s.io/client-go@v0.0.0-20191016111102-bec269661e48/tools/reference/ref.go:59 +0x116
k8s.io/client-go/tools/record.(*recorderImpl).generateEvent(0xc00017ff80, 0x176edc0, 0x0, 0x0, 0xbfb6dc9180116d97, 0x76e438a, 0x21d3d20, 0x1572ce5, 0x7, 0x157844f, ...)
	/mnt/go/pkg/mod/k8s.io/client-go@v0.0.0-20191016111102-bec269661e48/tools/record/event.go:291 +0x5d
k8s.io/client-go/tools/record.(*recorderImpl).Event(0xc00017ff80, 0x176edc0, 0x0, 0x1572ce5, 0x7, 0x157844f, 0xe, 0xc00019e1c0, 0xd2)
	/mnt/go/pkg/mod/k8s.io/client-go@v0.0.0-20191016111102-bec269661e48/tools/record/event.go:313 +0xc2
volcano.sh/volcano/pkg/controllers/job.(*Controller).initiateJob(0xc00045e000, 0x0, 0x0, 0x0, 0x0)
	/mnt/go/src/volcano.sh/volcano/pkg/controllers/job/job_controller_actions.go:152 +0x464
volcano.sh/volcano/pkg/controllers/job.(*Controller).syncJob(0xc00045e000, 0xc0009ba060, 0xc00113e470, 0x0, 0x0)
	/mnt/go/src/volcano.sh/volcano/pkg/controllers/job/job_controller_actions.go:195 +0x3d7
volcano.sh/volcano/pkg/controllers/job/state.(*pendingState).Execute(0xc000bcc000, 0x1572c21, 0x7, 0x2c, 0xc00048bd10)
	/mnt/go/src/volcano.sh/volcano/pkg/controllers/job/state/pending.go:54 +0xbc
volcano.sh/volcano/pkg/controllers/job.(*Controller).processNextReq(0xc00045e000, 0xc000000001, 0x1582200)
	/mnt/go/src/volcano.sh/volcano/pkg/controllers/job/job_controller.go:338 +0x8d8
volcano.sh/volcano/pkg/controllers/job.(*Controller).worker(0xc00045e000, 0xc000000001)
	/mnt/go/src/volcano.sh/volcano/pkg/controllers/job/job_controller.go:260 +0xaa
volcano.sh/volcano/pkg/controllers/job.(*Controller).Run.func1.1()
	/mnt/go/src/volcano.sh/volcano/pkg/controllers/job/job_controller.go:242 +0x31
k8s.io/apimachinery/pkg/util/wait.JitterUntil.func1(0xc0003e07a0)
	/mnt/go/pkg/mod/k8s.io/apimachinery@v0.16.9-beta.0/pkg/util/wait/wait.go:152 +0x5e
k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc000c79fa0, 0x3b9aca00, 0x0, 0x1, 0x0)
	/mnt/go/pkg/mod/k8s.io/apimachinery@v0.16.9-beta.0/pkg/util/wait/wait.go:153 +0xf8
k8s.io/apimachinery/pkg/util/wait.Until(...)
	/mnt/go/pkg/mod/k8s.io/apimachinery@v0.16.9-beta.0/pkg/util/wait/wait.go:88
volcano.sh/volcano/pkg/controllers/job.(*Controller).Run.func1(0xc00045e000, 0x0, 0x1)
	/mnt/go/src/volcano.sh/volcano/pkg/controllers/job/job_controller.go:240 +0x7b
created by volcano.sh/volcano/pkg/controllers/job.(*Controller).Run
	/mnt/go/src/volcano.sh/volcano/pkg/controllers/job/job_controller.go:239 +0x341
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x1178625]

goroutine 431 [running]:
k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
	/mnt/go/pkg/mod/k8s.io/apimachinery@v0.16.9-beta.0/pkg/util/runtime/runtime.go:55 +0x105
panic(0x13c30e0, 0x21ba5c0)
	/usr/local/go/src/runtime/panic.go:679 +0x1b2
volcano.sh/volcano/pkg/apis/batch/v1alpha1.(*Job).GetObjectKind(0x0, 0x17c7ac0, 0x0)
	<autogenerated>:1 +0x5
k8s.io/client-go/tools/reference.GetReference(0xc0002f5570, 0x176edc0, 0x0, 0x4dbf0b, 0x7f8eda6fc7e0, 0xc000c78918)
	/mnt/go/pkg/mod/k8s.io/client-go@v0.0.0-20191016111102-bec269661e48/tools/reference/ref.go:59 +0x116
k8s.io/client-go/tools/record.(*recorderImpl).generateEvent(0xc00017ff80, 0x176edc0, 0x0, 0x0, 0xbfb6dc9180116d97, 0x76e438a, 0x21d3d20, 0x1572ce5, 0x7, 0x157844f, ...)
	/mnt/go/pkg/mod/k8s.io/client-go@v0.0.0-20191016111102-bec269661e48/tools/record/event.go:291 +0x5d
k8s.io/client-go/tools/record.(*recorderImpl).Event(0xc00017ff80, 0x176edc0, 0x0, 0x1572ce5, 0x7, 0x157844f, 0xe, 0xc00019e1c0, 0xd2)
	/mnt/go/pkg/mod/k8s.io/client-go@v0.0.0-20191016111102-bec269661e48/tools/record/event.go:313 +0xc2
volcano.sh/volcano/pkg/controllers/job.(*Controller).initiateJob(0xc00045e000, 0x0, 0x0, 0x0, 0x0)
	/mnt/go/src/volcano.sh/volcano/pkg/controllers/job/job_controller_actions.go:152 +0x464
volcano.sh/volcano/pkg/controllers/job.(*Controller).syncJob(0xc00045e000, 0xc0009ba060, 0xc00113e470, 0x0, 0x0)
	/mnt/go/src/volcano.sh/volcano/pkg/controllers/job/job_controller_actions.go:195 +0x3d7
volcano.sh/volcano/pkg/controllers/job/state.(*pendingState).Execute(0xc000bcc000, 0x1572c21, 0x7, 0x2c, 0xc00048bd10)
	/mnt/go/src/volcano.sh/volcano/pkg/controllers/job/state/pending.go:54 +0xbc
volcano.sh/volcano/pkg/controllers/job.(*Controller).processNextReq(0xc00045e000, 0xc000000001, 0x1582200)
	/mnt/go/src/volcano.sh/volcano/pkg/controllers/job/job_controller.go:338 +0x8d8
volcano.sh/volcano/pkg/controllers/job.(*Controller).worker(0xc00045e000, 0xc000000001)
	/mnt/go/src/volcano.sh/volcano/pkg/controllers/job/job_controller.go:260 +0xaa
volcano.sh/volcano/pkg/controllers/job.(*Controller).Run.func1.1()
	/mnt/go/src/volcano.sh/volcano/pkg/controllers/job/job_controller.go:242 +0x31
k8s.io/apimachinery/pkg/util/wait.JitterUntil.func1(0xc0003e07a0)
	/mnt/go/pkg/mod/k8s.io/apimachinery@v0.16.9-beta.0/pkg/util/wait/wait.go:152 +0x5e
k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc000c79fa0, 0x3b9aca00, 0x0, 0x1, 0x0)
	/mnt/go/pkg/mod/k8s.io/apimachinery@v0.16.9-beta.0/pkg/util/wait/wait.go:153 +0xf8
k8s.io/apimachinery/pkg/util/wait.Until(...)
	/mnt/go/pkg/mod/k8s.io/apimachinery@v0.16.9-beta.0/pkg/util/wait/wait.go:88
volcano.sh/volcano/pkg/controllers/job.(*Controller).Run.func1(0xc00045e000, 0x0, 0x1)
	/mnt/go/src/volcano.sh/volcano/pkg/controllers/job/job_controller.go:240 +0x7b
created by volcano.sh/volcano/pkg/controllers/job.(*Controller).Run
	/mnt/go/src/volcano.sh/volcano/pkg/controllers/job/job_controller.go:239 +0x341

```

### PR DESCRIPTION
**Pull request by k82cn**
fix #890

### CODE REVIEW

**Comment by hzxuzhonghu** [comment]
/lgtm

the job nil issue will be in a separate pr.

File: pkg/controllers/job/job_controller_actions.go [hunk 1]
```diff
@@ -584,6 +584,7 @@ func (cc *Controller) initJobStatus(job *batch.Job) (*batch.Job, error) {
 
 	job.Status.State.LastTransitionTime = metav1.Now()
 	job.Status.State.Phase = batch.Pending
+	job.Status.State.LastTransitionTime = metav1.Now()
```
Comment (hzxuzhonghu) [inline pkg/controllers/job/job_controller_actions.go:587]: we have set it in L585

Comment (hzxuzhonghu) [reply pkg/controllers/job/job_controller_actions.go:587]: I must forget release 0.4.1

Comment (k82cn) [reply pkg/controllers/job/job_controller_actions.go:587]: let's have a 0.4.1 release this month :)

**Pull request closed without merging**